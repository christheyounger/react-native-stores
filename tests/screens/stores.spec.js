import 'isomorphic-fetch';
import React from 'react';
import renderer from 'react-test-renderer';
import StoresList from '../../app/screens/stores';

jest.mock('ScrollView', () => require.requireMock('ScrollViewMock'))

describe('screen.stores', function() {
    var navigation, wrapper;
    const storesList = [{id: 1, name: 'first store', branches: []}];
    beforeEach(() => {
        navigation = {
            navigate: jest.fn(),
        };
        wrapper = renderer.create(
            <StoresList
                navigation={navigation}
                getStores={jest.fn()}
                loading={false}
                stores={storesList}
            />
        );
    })
    it('shows loader', () => {
        wrapper = renderer.create(
            <StoresList
                navigation={navigation}
                getStores={jest.fn()}
                loading
                stores={storesList}
            />
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    })
    test('renders correctly', function() {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
    test('navigates to details', () => {
        wrapper.root.findByType('Text').props.onPress();
        expect(navigation.navigate).toHaveBeenCalled();
    })
});
