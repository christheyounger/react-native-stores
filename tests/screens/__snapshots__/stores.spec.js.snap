// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen.stores renders correctly 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Object {
          "branches": Array [],
          "id": 1,
          "name": "first store",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <Text
            onPress={[Function]}
          >
            - 
            first store
          </Text>
          <View
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            style={
              Object {
                "paddingLeft": 10,
              }
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`screen.stores shows loader 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
